src/Components/Components/test/Rendering/HtmlRendererTests.cs

```diff
@@ -374,15 +376,30 @@ public void RenderComponent_ElementRefsNoops()
                 rtb.CloseElement();
             })).BuildServiceProvider();
 
-            var htmlRenderer = new HtmlRenderer(serviceProvider, _encoder);
+            var htmlRenderer = new HtmlRenderer(serviceProvider, _encoder, _dispatcher);
 
             // Act
-            var result = htmlRenderer.RenderComponent<TestComponent>(ParameterCollection.Empty);
+            var result = GetResult(_dispatcher.Invoke(() => htmlRenderer.RenderComponent<TestComponent>(ParameterCollection.Empty)));
 
             // Assert
             Assert.Equal(expectedHtml, result);
         }
 
+        private IEnumerable<string> GetResult(Task<IEnumerable<string>> task)
+        {
+            Assert.True(task.IsCompleted);
+            if (task.IsCompletedSuccessfully)
+            {
+                return task.Result;
+            }
+            else
+            {
+                ExceptionDispatchInfo.Capture(task.Exception).Throw();
+                // We never get here
+                return null;
```

Comment:
Could this line be a `throw new InvalidOperationException("This line should never be hit");` instead then?